stempy/electron.cpp

```diff
@@ -166,27 +169,51 @@ std::vector<uint32_t> maximalPoints(
 
 template <typename InputIt>
 std::vector<std::vector<uint32_t>> electronCount(InputIt first, InputIt last,
-                                                 int scanRows, int scanColumns,
                                                  Image<double>& darkReference,
                                                  double backgroundThreshold,
-                                                 double xRayThreshold)
+                                                 double xRayThreshold,
+                                                 int scanRows, int scanColumns)
 {
+  if (first == last) {
+    std::ostringstream msg;
+    msg << "No blocks to read!";
+    throw std::invalid_argument(msg.str());
+  }
+
+  // If we haven't been provided with rows and columns, try the header.
+  if (scanRows == 0 || scanColumns == 0) {
+    scanRows = first->header.scanRows;
+    scanColumns = first->header.scanColumns;
+  }
+
+  // Raise an exception if we still don't have valid rows and columns
+  if (scanRows <= 0 || scanColumns <= 0) {
+    std::ostringstream msg;
+    msg << "No scan image size provided.";
+    throw std::invalid_argument(msg.str());
+  }
+
+  std::cout << scanRows << std::endl;
```

Comment:
Should we eliminate debug messages?

Reply:
I don't think it hurts having a few at this stage.

Reply:
I will remove them, I didn't mean to leave them in!